### PR TITLE
Allow icc compiler

### PR DESCRIPTION
**Pull request by meena-vyas**
Allow ICC compiler too.

This is built on top of pull request https://github.com/openssl/openssl/pull/4249 so I could test on Solaris.

### CODE REVIEW

File: Configure [hunk 1]
```diff
@@ -1323,8 +1323,8 @@ if (defined($config{api})) {
 if ($strict_warnings)
 	{
 	my $wopt;
-	die "ERROR --strict-warnings requires gcc or clang"
-            unless $ecc eq 'gcc' || $ecc eq 'clang';
+	die "ERROR --strict-warnings requires gcc , clang or icc"
```
Comment (levitte) [inline Configure]: Nit pick: remove the space before the comma

**Comment by levitte** [comment]
Squashed and merged.  a610934c3b

I edited the commit message a bit:

```
Allow --strict-warnings with the icc compiler as well
```

**Comment by mattcaswell** [comment]
I'm not sure why we merged this. The implication is that we are going to support icc with `--strict-warnings` and fix all the issues. But #4248 suggests that might not be desirable.

**Comment by richsalz** [comment]
Doesn't this change *allow* but not *require* --strict-warnings?

**Comment by mattcaswell** [comment]
It does, but without #4248 it doesn't actually work with icc which just invites issues to be raised.

**Comment by levitte** [comment]
I don't see a reason why we can't have this, still...  It's a community contribution, is `Configure` excempt from that?

**Comment by richsalz** [comment]
INSTALL says the right text about --strict-warnings; it should stay.

**Comment by paulidale** [comment]
Allowing another mature compiler to critique the code base is unlikely to be a retrograde step, it only has to find a problem or two to be worthwhile.

As for #4248 either the issue there will have to be addressed or a work around found (disable that warning?).  I don't see it as a major impediment but any noise from it should be avoided.

**Pull request closed without merging**